src/implementation/Client/HTTPClient/HTTPClient.ts

```diff
@@ -151,6 +151,12 @@ export default class HTTPClient implements IClient {
 
     if (!params?.headers) {
```

Comment:
This adds the API token to headers if they are missing, but if the caller has passed headers, we also need to append the token.